mapflow/functional/layer_utils.py

```diff
@@ -501,40 +501,54 @@ def download_results_callback(self, response: QNetworkReply, processing) -> None
-        output_path = os.path.join(self.dlg.outputDirectory.text(), processing.id_)
-        extension = '.gpkg'
-        if os.path.exists(output_path + extension):
+        output_path = Path(self.dlg.outputDirectory.text(), processing.id_).with_suffix(".gpkg")
+        if output_path.exists():
             count = 1
-            while os.path.exists(output_path + f'({count})' + extension):
+            while output_path.with_stem(processing.id_ + f"_{count}").exists():
                 count += 1
-            output_path += f'({count})' + extension
-        else:
-            output_path += extension
+            output_path = output_path.with_stem(processing.id_ + f"_{count}")
         transform = self.project.transformContext()
         # Layer creation options for QGIS 3.10.3+
         write_options = QgsVectorFileWriter.SaveVectorOptions()
         write_options.layerOptions = ['fid=id']
-        with open(os.path.join(self.temp_dir, os.urandom(32).hex()), mode='wb+') as f:
-            f.write(response.readAll().data())
+        with open(Path(self.temp_dir, os.urandom(32).hex()), mode='wb+') as f:
+            response_data = response.readAll().data()
+            f.write(response_data)
             f.seek(0)
             layer = QgsVectorLayer(f.name, '', 'ogr')
             # V3 returns two additional str values but they're not documented, so just discard them
             error, msg, *_ = QgsVectorFileWriter.writeAsVectorFormatV3(
                 layer,
-                output_path,
+                str(output_path),
                 transform,
                 write_options
             )
         if error:
-            self.message_bar.pushWarning(self.tr("Error"),
-                                         self.tr('Failed to save results to file. '
-                                                 'Error code: {code}. Message: {message}').format(code=error,
-                                                                                                  message=msg))
-            return
+            self.message_bar.pushWarning(self.tr("Warning"),
+                                self.tr('Failed to save results to GeoPackage. '
+                                        'Error code: {code}. Message: {message}. ' 
+                                        'File will be saved as GeoJSON instead.').format(code=error, message=msg))
+            # Delete empty GeoPackage
+            output_path.unlink()
+            # Save as GeoJSON instead
+            output_path = output_path.with_suffix(".geojson")
+            if output_path.exists():
+                count = 1
+                while output_path.with_stem(processing.id_ + f"_{count}").exists():
+                    count += 1
+                output_path = output_path.with_stem(processing.id_ + f"_{count}")
+            try:
+                with open(str(output_path), mode='wb+') as f:
+                        f.write(response_data)
+                        f.seek(0)
```

Comment:
Remove seek(0) here. 

seek(0) was added above so that when layer is created from this open file, it would start reading it from the start, and here we just close the file after this